receiver/kubeletstatsreceiver/internal/kubelet/network.go

```diff
@@ -10,18 +10,20 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/kubeletstatsreceiver/internal/metadata"
 )
 
-type getNetworkDataFunc func(s *stats.NetworkStats) (rx *uint64, tx *uint64)
+type getNetworkDataFunc func(s *stats.InterfaceStats) (rx *uint64, tx *uint64)
 
 func addNetworkMetrics(mb *metadata.MetricsBuilder, networkMetrics metadata.NetworkMetrics, s *stats.NetworkStats, currentTime pcommon.Timestamp) {
 	if s == nil {
 		return
 	}
 
-	recordNetworkDataPoint(mb, networkMetrics.IO, s, getNetworkIO, currentTime)
-	recordNetworkDataPoint(mb, networkMetrics.Errors, s, getNetworkErrors, currentTime)
+	for i := range s.Interfaces {
+		recordNetworkDataPoint(mb, networkMetrics.IO, &s.Interfaces[i], getNetworkIO, currentTime)
+		recordNetworkDataPoint(mb, networkMetrics.Errors, &s.Interfaces[i], getNetworkErrors, currentTime)
+	}
```

Comment:
If `s.Interfaces` contains more than just `eth0`, then this will cause the receiver to produce more datapoints. While the `interface` attribute was always present, it has historically only ever had one value.  Now it will have more, so the cardinality of the metric effectively increases.

I believe we need to put this behind a feature gate or a config option so that users are not surprised.

Reply:
I see, thank's for the context. A feature gate sounds good to me.

Reply:
Thinking of it again, I realize that maybe this change cannot be a drop-in replacement.
Some users might already be happy with the default network metrics, so adding metrics for all the interfaces by-default might be problematic.
I think making this functionality configurable makes more sense (compared to going with a feature flag).

So instead of the feature flag I think introducing 2 new settings would provide better flexibility here:
```
add_node_interfaces_metrics: true|false
add_pod_interface_metrics: true|false
```

The default behavior will continue to be as is today, collecting only metrics for the default network interface. By enabling the "interface_metrics" settings users will be explicitly making the receiver to include the additional metrics for all the interfaces.

Also providing the node/pod level granularity will be useful since users might want to avoid collecting all Pod's interfaces metrics when Pods are in `hostNetwork: true` mode (as mentioned at https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30196#issuecomment-1894299580). 

@TylerHelmuth @povilasv let me know what you think here.

Reply:
Ok! My only concern is about the `hostNetwork` case but eventually if a Pod is plugged to the host network it makes sense to observe "its" traffic anyways.
I have put the change behind a feature flag. Let me know you think folks.

Reply:
I'm more in favor of the config option, although I am curious when the actual behavior would be in a standard cluster.  Will most have other interfaces to collector metrics from?

@dmitryax @jinja2 what are your thoughts?